### ISSUE TITLE
Use instruction count as metric during benchmarking

### ISSUE THREAD

**Issue by bjorn3** [bjorn3/rustc_codegen_cranelift#803]
It is more stable than wall-time.

**Comment by philipc** [comment]
Best not to only use instruction counts though, because they don't tell the whole story. I've had situations with gimli whether the instruction counts are similar but the execution times are quite different.

**Comment by bjorn3** [comment]
Yes, I still want to measure wall time. Currently I only use wall time, however it is so noisy on my system (~5% variation) that it is hard to tell the difference between cg_clif and cg_llvm without running a benchmark many times. Instruction count is much more stable.

> I've had situations with gimli whether the instruction counts are similar but the execution times are quite different.

A lot of time is spent in other parts than cg_clif, like rustc and cranelift. This means that optimizations of cg_clif itself are unlikely to cause a big mismatch between instruction count and wall time changes.

**Comment by bjorn3** [comment]
Using `schedtool -B -e ./raytracer_cg_clif` may reduce noise. The batch scheduler mode uses longer time slices, which may result in less noise.

**Comment by bjorn3** [comment]
https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/ might be useful to report this once it is being measured.
